src/NRules/NRules.RuleModel/NamedLambdaExpression.cs

```diff
@@ -0,0 +1,74 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+
+namespace NRules.RuleModel
+{
+    /// <summary>
+    /// Lambda Expression with a name used in constructing/building an aggregator.
+    /// </summary>
+    public class NamedLambdaExpression : INamedExpression<LambdaExpression>
```

Comment:
Overall, I was thinking to use the following constructs:
* Primitives, like `KVP<string, LambdaExpression>` in builders and static factories.
* ExpressionCollection and NamedExpressionElement in the canonical model
* IAggregateExpression in the executable model (just add a Name to it, see my other comment)

Then, this NamedLambdaExpression is not needed, likewise INamedExpression and NamedAggregateExpression. Want to reduce coupling between layers as much as possible.